java/core/src/main/java/co/worklytics/psoxy/rules/msft/PrebuiltSanitizerRules.java

```diff
@@ -275,6 +292,219 @@ public class PrebuiltSanitizerRules {
+        .build();
+
+    static final Endpoint MS_TEAMS_CHATS_MESSAGES =  Endpoint.builder()
+        .pathTemplate(MS_TEAMS_PATH_TEMPLATES_CHATS_MESSAGES)
+        .transform(Transform.Redact.builder()
+            .jsonPath("$..user.displayName")
+            .jsonPath("$..value[*].body.content")
+            .jsonPath("$..value[*].attachments")
+            .jsonPath("$..value[*].mentions[*].mentionText")
+            .jsonPath("$..value[*].eventDetail.teamDescription")
+            .jsonPath("$..value[*].eventDetail.chatDisplayName")
+            .build())
+        .allowedQueryParams(List.of("$select","$top","$skipToken","$filter", "$orderBy", "$count", "$expand", "$format", "$search", "$skip"))
+        .build();
+
+    static final Endpoint MS_TEAMS_COMMUNICATIONS_CALLS = Endpoint.builder()
+        .pathTemplate(MS_TEAMS_PATH_TEMPLATES_COMMUNICATIONS_CALLS)
+        .transform(Transform.Redact.builder()
+            .jsonPath("$..displayName")
+            .build())
+        .allowedQueryParams(List.of("$select","$top","$expand"))
+        .build();
+
+    static final Endpoint MS_TEAMS_COMMUNICATIONS_CALL_RECORDS = Endpoint.builder()
+        .pathRegex(MS_TEAMS_PATH_TEMPLATES_COMMUNICATIONS_CALL_RECORDS_REGEX)
+        .transform(Transform.Redact.builder()
+            .jsonPath("$..user.displayName")
+            .jsonPath("$..reflexiveIPAddress")
+            .jsonPath("$..relayIPAddress")
+            .jsonPath("$..ipAddress")
+            .jsonPath("$..subnet")
+            .jsonPath("$..macAddress")
+            .jsonPath("$..caller.name")
+            .jsonPath("$..callee.name")
+            .jsonPath("$..captureDeviceName")
+            .jsonPath("$..renderDeviceName")
+            .build())
+        .allowedQueryParams(List.of("$select","$expand"))
+        .build();
+
+    static final Endpoint MS_TEAMS_COMMUNICATIONS_CALL_RECORDS_GET_DIRECT_ROUTING_CALLS = Endpoint.builder()
+        .pathTemplate(MS_TEAMS_PATH_TEMPLATES_COMMUNICATIONS_CALL_RECORDS_GET_DIRECT_ROUTING_CALLS)
+        .transform(Transform.Redact.builder()
+            .jsonPath("$..value[*].userPrincipalName")
+            .jsonPath("$..value[*].userDisplayName")
+            .jsonPath("$..value[*].callerNumber")
+            .jsonPath("$..value[*].calleeNumber")
+            .build())
+        .allowedQueryParams(List.of("$skip"))
+        .build();
+
+    static final Endpoint MS_TEAMS_COMMUNICATIONS_CALL_RECORDS_GET_PSTN_CALLS = Endpoint.builder()
+        .pathTemplate(MS_TEAMS_PATH_TEMPLATES_COMMUNICATIONS_CALL_RECORDS_GET_PSTN_CALLS)
+        .transform(Transform.Redact.builder()
+            .jsonPath("$..value[*].userPrincipalName")
+            .jsonPath("$..value[*].userDisplayName")
+            .jsonPath("$..value[*].callerNumber")
+            .jsonPath("$..value[*].calleeNumber")
+            .build())
+        .allowedQueryParams(List.of("$skip"))
+        .build();
+
+    static final Endpoint MS_TEAMS_USERS =  Endpoint.builder()
```

Comment:
There is already a `DIRECTORY_USERS` endpoint for the same

Reply:
Done.